sdk/devcenter/azure-developer-devcenter/src/main/java/com/azure/developer/devcenter/EnvironmentsClientBuilder.java

```diff
@@ -0,0 +1,338 @@
+import com.azure.core.annotation.ServiceClientBuilder;
+import com.azure.core.client.traits.ConfigurationTrait;
+import com.azure.core.client.traits.HttpTrait;
+import com.azure.core.client.traits.TokenCredentialTrait;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.HttpPipelinePosition;
+import com.azure.core.http.policy.AddDatePolicy;
+import com.azure.core.http.policy.AddHeadersFromContextPolicy;
+import com.azure.core.http.policy.AddHeadersPolicy;
+import com.azure.core.http.policy.BearerTokenAuthenticationPolicy;
+import com.azure.core.http.policy.CookiePolicy;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.http.policy.HttpLoggingPolicy;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.HttpPolicyProviders;
+import com.azure.core.http.policy.RequestIdPolicy;
+import com.azure.core.http.policy.RetryOptions;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.http.policy.UserAgentPolicy;
+import com.azure.core.util.ClientOptions;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.builder.ClientBuilderUtil;
+import com.azure.core.util.serializer.JacksonAdapter;
+import com.azure.developer.devcenter.implementation.DevCenterClientImpl;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/** A builder for creating a new instance of the EnvironmentsClient type. */
+@ServiceClientBuilder(serviceClients = {EnvironmentsClient.class, EnvironmentsAsyncClient.class})
+public final class EnvironmentsClientBuilder
+        implements HttpTrait<EnvironmentsClientBuilder>,
+                ConfigurationTrait<EnvironmentsClientBuilder>,
+                TokenCredentialTrait<EnvironmentsClientBuilder> {
+    @Generated private static final String SDK_NAME = "name";
+
+    @Generated private static final String SDK_VERSION = "version";
+
+    @Generated private static final String[] DEFAULT_SCOPES = new String[] {"https://devcenter.azure.com/.default"};
+
+    @Generated
+    private final Map<String, String> properties = CoreUtils.getProperties("azure-developer-devcenter.properties");
+
+    @Generated private final List<HttpPipelinePolicy> pipelinePolicies;
+
+    /** Create an instance of the EnvironmentsClientBuilder. */
+    @Generated
+    public EnvironmentsClientBuilder() {
+        this.pipelinePolicies = new ArrayList<>();
+    }
+
+    /*
+     * The HTTP pipeline to send requests through.
+     */
+    @Generated private HttpPipeline pipeline;
+
+    /** {@inheritDoc}. */
+    @Generated
+    @Override
+    public EnvironmentsClientBuilder pipeline(HttpPipeline pipeline) {
+        this.pipeline = pipeline;
```

Comment:
For @weidongxu-microsoft, for `pipeline` and `httpClient` we should update this to copy what the hand-written builders do with logging a debug message if the HttpPipeline, or HttpClient, is set to null after being configured as there is a chance that it is a bug in user code:

```java
if (this.pipeline != null && pipeline == null) {
    logger.info("HttpPipeline is being set to 'null' when it was previously configured.");
}
```